benchmarks/tool/compile_benchmarks.dart

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env dart
+
+import 'dart:io' show exit, Platform, Process, Directory, ProcessResult;
+
+import 'package:args/args.dart' show ArgParser;
+import 'package:path/path.dart' as path;
+import 'package:pool/pool.dart' show Pool;
+
+Future<void> main(List<String> args) async {
+  final argParser = ArgParser()
+    ..addOption('target',
+        mandatory: false, defaultsTo: 'aot,exe,jit,js,js-production')
+    ..addOption('jobs', abbr: 'j', mandatory: false);
+
+  final parsedArgs = argParser.parse(args);
+
+  var jobs = Platform.numberOfProcessors;
+  if (parsedArgs['jobs'] != null) {
+    jobs = int.parse(parsedArgs['jobs']!);
+  }
+
+  final targets = <Target>{};
+  for (final targetStr in parsedArgs['target'].split(',')) {
+    switch (targetStr) {
+      case 'aot':
+        targets.add(aotTarget);
+        break;
+
+      case 'exe':
+        targets.add(exeTarget);
+        break;
+
+      case 'jit':
+        targets.add(jitTarget);
+        break;
+
+      case 'js':
+        targets.add(jsTarget);
+        break;
+
+      case 'js-production':
+        targets.add(jsProductionTarget);
+        break;
+
+      default:
+        print(
+            'Unsupported target: $targetStr. Supported targets: aot, exe, jit, js, js-production');
+        exit(1);
+    }
+  }
+
+  // Arg list is immutable, clone it
+  final sourceFiles = List.from(parsedArgs.rest);
+
+  if (sourceFiles.isEmpty) {
+    // Compile all files in bin/
+    final dir = Directory('bin');
+    for (final dirFile in dir.listSync(recursive: false)) {
+      // Compiler hangs (instead of failing) when passed an .exe, so only try
+      // to compile Dart files
```

Comment:
Well yes, but logically it only makes sense to compile .dart files, it doesn't make sense for this program to compile random files regardless of whether the compiler hangs or not